### DISCUSSION TITLE
I am wondering how to route to a specific product id in nuxt3 [router, nuxt, vue, restAPI]

### DISCUSSION THREAD

**Discussion by appleofeden110** [nuxt/nuxt#22545] [Questions] [answered]
I am trying to fetch from my Flask REST API some test products. Here is the code:

```
<template>
    <div>
        <div class="grid grid-cols-4 gap-5">
            <div v-for="p in data">
                <nuxt-link :to="/product/`${p.id}`">{{ p.name }}</nuxt-link>
            </div>
        </div>
    </div>

    
</template>

<script setup>
const { data } = await useFetch('http://127.0.0.1:5000/product'); 


</script>

<style scoped>

</style>
```

I am trying to make an automatic grid of products which take id's of every one of the products and directing to the page which will show an individual description of it (I am learning). 

tried to use @to or just to, however both of them either showed just what I've written in the link, or didn't even work as an href.

Any ideas how can I solve that problem? Been hitting my head against the table for few days already.

**Comment by WailanTirajoh** [comment, marked as answer]
The `/product/` looks like regex, so it doesn't show error with v-bind to. If you mark attributes with ":" the value inside it is Javascript expression, maybe the Javascript thinks /product/ is a regex so it doesn't show any sign of error.

try this instead.
```:to="`/product/${p.id}`"```

It's also good practice to add unique key on loop. The final code maybe look like this

```
<template>
    <div>
        <div class="grid grid-cols-4 gap-5">
            <div v-for="p in data" :key="p.id">
                <nuxt-link :to="`/product/${p.id}`">{{ p.name }}</nuxt-link>
            </div>
        </div>
    </div>
</template>
```

Also nuxt useFetch composable has second params for baseUrl. 

```
<script setup>
const { data } = await useFetch('/product', {
  baseUrl: 'http://127.0.0.1:5000'
}); 
</script>
```

**Reply by appleofeden110** [reply]
Thanks man! Honestly I got so frustrated with Nuxt and Vue that I switched to the Sveltekit and Svelte. Not gonna lie, it's the best I could do as I managed to do a lot more a lot faster. To each their own, I suppose. But still thanks a lot for the answer! I see you had some upvotes so don't think that your answer wasn't helpful!
